leases/models.py

```diff
@@ -218,8 +228,12 @@ def clean(self):
             existing_leases = existing_leases.exclude(pk=self.pk)
         if existing_leases.exists():
             raise ValidationError(_("WinterStoragePlace already has a lease"))
-        if not self.place.is_active and creating:
+        if creating and self.place and not self.place.is_active:
             raise ValidationError(_("Selected place is not active"))
+
+        if self.place and self.area:
+            raise ValidationError(_("Lease cannot have both place and area assigned"))
+
```

Comment:
Should we also check that either place or area is set?

Reply:
Yes! Good catch. 👍